deepcell/model_zoo/panopticnet.py

```diff
@@ -48,21 +48,41 @@
 
 
 def __merge_temporal_features(feature, mode='conv', feature_size=256, frames_per_batch=1):
-    if mode == 'conv':
+    """
+
+    Args:
+        feature: Input layer
+        mode (str, optional): Choose from {'conv','lstm','gru'}. Defaults to 'conv'.
+        feature_size (int, optional): Defaults to 256.
+        frames_per_batch (int, optional): Defaults to 1.
+
+    Raises:
+        ValueError: Mode not 'conv', 'lstm' or 'gru'
+
+    Returns:
+        Same as feature: Output of layer following merge of temporal features
+    """
+
+    acceptable_modes = {'conv', 'lstm', 'gru'}
+    if mode.lower() not in acceptable_modes:
+        raise ValueError('Mode {} not supported. Please choose from {}.'.format(
+            mode.lower(), str(acceptable_modes)))
+
+    if mode.lower() == 'conv':
         x = Conv3D(feature_size,
                    (frames_per_batch, 3, 3),
                    strides=(1, 1, 1),
                    padding='same',
                    )(feature)
         x = BatchNormalization(axis=-1)(x)
         x = Activation('relu')(x)
-    elif mode == 'lstm':
+    elif mode.lower() == 'lstm':
         x = ConvLSTM2D(feature_size,
                        (3, 3),
                        padding='same',
                        activation='relu',
                        return_sequences=True)(feature)
-    elif mode == 'gru':
+    elif mode.lower() == 'gru':

```

Comment:
We still need an else block or else `x` is not defined.

Reply:
I'm not totally sure how `x` should be defined in the else case. Would you want it to return `feature` without any modification? I thought that by putting in `raise ValueError` that we would avoid getting to the condition where else is triggered.

Reply:
We would avoid that for now, but it is just an opening for a `NullPointerError`. This is a coding best-practice issue.  If `mode is None`, maybe we should just return the unmodified feature?